File: Level.2/위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    int clothes_size = clothes.size();
    map<string, int> clothes_map;
    for(int i = 0; i < clothes_size; ++i)
    {
        ++clothes_map[clothes[i][1]];
    }
    map<string, int>::iterator iter;
    for(iter = clothes_map.begin(); iter != clothes_map.end(); ++iter)
    {
        answer *= (iter->second + 1); 
    }
    
    answer -= 1;
    return answer;
}

File: JavaScript/Level.2/위장.js
function solution(clothes) {
    var answer = 1;
    let map = new Map();
    for (let clothe of clothes) {
        if (map.has(clothe[1])) {
            map.set(clothe[1], map.get(clothe[1]) + 1);
        } else {
            map.set(clothe[1], 1);
        }
    }
    for (let [key, value] of map) {
        answer *= value + 1;
    }
    answer -= 1;
    return answer;
}
